sunpycube/cube/NDCube.py

```diff
@@ -71,12 +69,8 @@ def plot(self, *args, **kwargs):
     def __getitem__(self, item):
         if item is None or (isinstance(item, tuple) and None in item):
             raise IndexError("None indices not supported")
-        return self._getitem_NDCube(item)
-
-    def _getitem_NDCube(self, item):
         data = self.data[item]
-        # wcs = self.wcs.wcs_slicer[item]
-        wcs = None
+        wcs = self.wcs.wcs_slicer[item]
```

Comment:
So the slicer should be a function.  Yes, I completely agree.  We should do that.

Also, could the augmentation stuff be "functionalised" as well so that you can always use an astropy.wcs object?  But perhaps there would have to be records attached to the wcs object to make the augmentation work?  And that would require a subclass?

Another idea is to have a function in an if statement in the ```NDCube.__init__``` that converts an astropy.wcs object to a sunpycube wcs object if the user supplies an astropy.wcs?  That way at least the user can supply the intuitive wcs object, i.e. the astropy.wcs one.